tb_plugin/torch_tb_profiler/profiler/data.py

```diff
@@ -118,12 +118,33 @@ def _preprocess_file(caches, trace_path):
                     # only replace the N/A without surrounding double quote
                     fout.write(re.sub(r'(?<!")N/A(?!")', "\"N/A\"", str_data))
                     trace_json = json.loads(fout.getvalue())
-
+                    logger.warning("Get JSONDecodeError: %s, Re-encode it to temp file" % e.msg)
+                    json_reencode = True
+
+        # work-around to remove the "Record Window End" events to avoid the huge end timestamp
+        event_list = trace_json["traceEvents"]
+        end_index = None
+        start_index = None
+        for i in reversed(range(len(event_list))):
```

Comment:
Is there not a better place to do this in some existing iteration to avoid extra load latency? Or is it fast enough?

Reply:
I guess it's quickly found since you search backwards.

Reply:
yes, the last two items will be the "Iteration Start: PyTorch Profiler" and "Record Window End" instant events.